File: one-by-one/11_example_launch_pytest/example_launch_pytest/twice.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import Int32

class TwiceNode(Node):
    def __init__(self):
        super().__init__("twice")
        self._pub = self.create_publisher(Int32, "dst", 10)
        self._sub = self.create_subscription(
            Int32,
            "src",
            lambda msg: self._pub.publish(Int32(data=msg.data * 2)),
            10
        )

def main(args=None):
    rclpy.init(args=args)
    node = TwiceNode()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: one-by-one/10_example_launch_testing_ament_cmake/src/twice.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/int32.hpp>

using Int32 = std_msgs::msg::Int32;

class TwiceNode : public rclcpp::Node {
public:
  TwiceNode() : Node("twice", "") {
    pub_ = this->create_publisher<Int32>("dst", 10);
    sub_ = this->create_subscription<Int32>(
        "src", 10, [this](const Int32::ConstSharedPtr src) -> void {
          auto output = Int32();
          output.data = src->data * 2;
          this->pub_->publish(output);
        });
  }

private:
  rclcpp::Subscription<Int32>::SharedPtr sub_;
  rclcpp::Publisher<Int32>::SharedPtr pub_;
};

int main(int argc, char *argv[]) {
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<TwiceNode>());
  rclcpp::shutdown();
  return 0;
}
